### PR TITLE
Upstream #470 - Adds ParamChangeProposal

### PR DESCRIPTION
**Pull request by boojamya**
This PR...

(Partially) Upstreams #470. 

Unlike in `v3`, the `ParamChangeProposal` function was already implemented. However, in this branch ('main'), it intakes a type rather than an actual json file. If necessary, the json file can be unmarshalled during specific test cases if a user prefers. 

Happy to adjust this if intaking a file to the `ParamChangeProposal` is a better flow. 

Also... (unrelated to the upstream)

I didn't think it was necessary to have both `chain_upgrade_test.go` and `chain_upgrade_ibc_test`. 
They seem to be testing the same thing, except the IBC version ran conformance before and after the upgrade.
I kept the IBC version.

### CODE REVIEW

**Comment by boojamya** [comment]
cc: @Reecepbcups 
As the OG author of #470 and #467, just want to make sure you see this. 

We will be branching new versions of our maintained branches off main (see issue: https://github.com/strangelove-ventures/interchaintest/issues/539)

If you have any concerns, let us know.

**Comment by Reecepbcups** [comment]
@boojamya Thanks! yea the unmarshing is a good point - I was writing it from an idea of making it easy to show examples of pre formatted json files ready for chain use. 

Unmarshing will work for this

File: chain/cosmos/genesis.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/icza/dyno"
+	"github.com/strangelove-ventures/interchaintest/v7/ibc"
+)
+
+func ModifyGenesisProposalTime(votingPeriod string, maxDepositPeriod string) func(ibc.ChainConfig, []byte) ([]byte, error) {
```
Comment (misko9) [inline chain/cosmos/genesis.go:11]: This works for gov modules that are on v1beta1.
Gov v1 will look like this, so maybe break out into two functions with a Gov v1beta1 vs v1 name?
```
dyno.Set(g, votingPeriod, "app_state", "gov", "params", "voting_period")
dyno.Set(g, maxDepositPeriod, "app_state", "gov", "params", "max_deposit_period")
dyno.Set(g, chainConfig.Denom, "app_state", "gov", "params", "min_deposit", 0, "denom")
```

Comment (boojamya) [reply chain/cosmos/genesis.go:11]: Ahh. Good eye!
Will take a pass at this.

Comment (boojamya) [reply chain/cosmos/genesis.go:12]: @misko9
Ok, added gov module version check.

**Review by misko9 — APPROVED** [review]
LGTM!

**Pull request merged**